From c1d5fab38d2eae94866b33967a5a8674b52e1a54 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 27 Aug 2020 17:54:53 +0200
Subject: [PATCH 1/1] efi_loader: remove duplicate image size check

The image size is checked in efi_load_pe(). Avoid checking it twice.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index d49145fc76..dcd3eec894 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1883,10 +1883,6 @@ efi_status_t EFIAPI efi_load_image(bool boot_policy,
 		if (ret != EFI_SUCCESS)
 			goto error;
 	} else {
-		if (!source_size) {
-			ret = EFI_LOAD_ERROR;
-			goto error;
-		}
 		dest_buffer = source_buffer;
 	}
 	/* split file_path which contains both the device and file parts */
-- 
2.28.0

